sdk/core/core/src/az_http_response_parser.c

```diff
@@ -298,3 +298,32 @@ az_http_response_get_body(az_span const self, az_pair * const p_last_header, az_
   AZ_RETURN_IF_FAILED(az_http_response_parser_get_body(&parser, body));
   return AZ_OK;
 }
+
+/**
+ * Get an HTTP header by name.
+ *
+ * Returns @AZ_ERROR_HTTP_NO_MORE_ITEMS
+ */
+AZ_NODISCARD az_result az_http_response_get_header_by_name(
+    az_span const self,
+    az_span const header_name,
+    az_span * const header_value) {
+  AZ_CONTRACT_ARG_NOT_NULL(header_value);
+
+  az_http_response_parser parser;
+  AZ_RETURN_IF_FAILED(az_http_response_parser_init(&parser, self));
+
+  { 
+    az_http_response_status_line status_line;
+    AZ_RETURN_IF_FAILED(az_http_response_parser_get_status_line(&parser, &status_line)); 
+  }
+
+  while (true) {
```

Comment:
I think it would be easier to read something like
```
do {
    result = get_next_header(...)
    if (....
       return ...
while (!NO_MORE_ITEMS)

return NOT_FOUND
```

Reply:
Could you provide an example what will be `if (…` ? Because we need to analyze errors as well and we can't just use `az_failed`

Reply:
add more comments.